src/object_store.cpp

```diff
@@ -517,6 +517,19 @@ void ObjectStore::verify_compatible_for_read_only(std::vector<SchemaChange> cons
     verify_no_errors<InvalidSchemaChangeException>(verifier, changes);
 }
 
+void ObjectStore::verify_compatible_for_read_only_alternative(std::vector<SchemaChange> const& changes)
+{
+    using namespace schema_change;
+    struct Verifier : SchemaDifferenceExplainer {
+        using SchemaDifferenceExplainer::operator();
+
+        void operator()(RemoveProperty) { }
```

Comment:
> Yes it is... That is the current realm-java's behavior. I have a discussion with @cmelchior before, we think it is better to keep this since it makes less sense to return an empty results when user query on a non-existing table than just throw earlier.

What's the argument for this behaving differently than the immutable mode? It's not clear to me why this mode should throw if the app's schema contains classes not present in the Realm when immutable mode doesn't.

> This is handled in another way in java, open a dynamic realm, download changes, then open the typed realm.

Sure, that's roughly how Cocoa implements its equivalent of "wait for initial remote data" as well. But what about using this schema mode _without_ having to wait for all remote changes to be downloaded? Requiring that the set of classes in the schema match prevents using it for this. See realm/realm-cocoa#5186 for a scenario where this would be useful.